aries_cloudagent/resolver/did_resolver.py

```diff
@@ -35,27 +29,30 @@ async def resolve(self, profile: Profile, did: Union[str, DID]) -> DIDDocument:
         """Retrieve did doc from public registry."""
         # TODO Cache results
         py_did = DID(did) if isinstance(did, str) else did
-        for resolver in self._match_did_to_resolver(py_did):
+        did = str(py_did)
```

Comment:
Similar to my other comment, we ultimately want a string that's been validated as being a real DID and we can do that without assigning to a variable:
```python
if isinstance(did, DID):
    did = str(did)
else:
    DID.validate(did)  # <- will raise an error if invalid
```

This simplification would extend into the `_match_did_to_resolver` method.

Reply:
_match_did_to_resolver is called with py_did, which uses DID to get the method as in comment above. how should it look?